### DISCUSSION TITLE
Can't get SD to initialize

### DISCUSSION THREAD

**Discussion by TheLardos** [earlephilhower/arduino-pico#2243] [General]
Hi,

I'm trying to get my SD card module initialized, but no matter what I try, I just can't get it to work.

I've written a simple code:

```
#include <SPI.h>
#include <SD.h>

// Definition der Pins
const int chipSelect = 1; // GP1 (CSn)

void setup() {
  Serial.begin(9600);
  while (!Serial) {
  }

  SPI.setRX(0); // GP0 als MISO
  SPI.setTX(3); // GP3 als MOSI
  SPI.setSCK(2); // GP2 als SCK
  SPI.begin();

  Serial.print("InitializeSD-Card...");
  if (!SD.begin(chipSelect)) {
    Serial.println("SD-card couldn't be initialized.");
    return;
  }
  Serial.println("successful");
}

void loop() {
}

```

I am using a Pimoroni tiny rp2040 (here is a photo of the pinout: 
![19278_5](https://github.com/earlephilhower/arduino-pico/assets/74902184/49b6faa3-3d89-471f-baf1-c4c0808cd1f4)
 )
I appreciate any help I can get on this.

Thanks in advance!
Luca

**Comment by earlephilhower** [comment]
Moved to a discussion since it's not a core issue.

Double check your wiring (i.e. GPx is not the same as your PCB pins) and ensure you're running a 3.3V capable adapter.  Some of those boards have onboard 5V regulators and I/O buffers (SD is 3.3V, but old Arduino boards were 5V) which could make them incompatible with the 3.3V pico.  You really want a board with nothing active on it.  I've even successfully used a microSD-to-fill-SD adapter with wires tacked on to the contacts, directly connected to the Pico GPIOs.

**Reply by TheLardos** [reply]
Thank you for your quick reply. I tested your method of directly soldering to an microSD-to-fill-SD adapter and this actually worked! So it has something to do with the module that I used..

Thank you for your help and sorry for putting this topic in the wrong category!

**Reply by earlephilhower** [reply]
No worries, glad to hear it's running.

For more "professional" use, I've had success with boards like this from the usual suspects:
![image](https://github.com/earlephilhower/arduino-pico/assets/11875/2f9f744d-5060-41fe-9f40-47984299aca1)

It's basically a straight-through wiring with only some Rs and Cs to better meet the SD specs for slew/etc.

**Reply by TheLardos** [reply]
That's exactly what I just ordered! Thank you for your support!
